### PR TITLE
Validations and model specs

### PR DESCRIPTION
**Pull request by alicemirigo92**
1. Use validations for models.
2. Write unit tests for models.

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -3,6 +3,10 @@ class Post < ApplicationRecord
   has_many :comments, dependent: :destroy
   has_many :likes, dependent: :destroy
   after_save :update_post_counter
+  validates :title, presence: true, length: { maximum: 250 }
+
```
Comment (zainsadaqat) [inline app/models/post.rb:8]: ### [REQUIREMENT]
Good job for adding validation for the title, likescounter, and commentcounter. However, there's one validation is missing which is about "Title must not exceed 250 characters." as it's part of project's requirements.

Screenshot 2023-05-13 at 5 14 03 AM

**Review by zainsadaqat — CHANGES_REQUESTED** [review]
# Required Changes 🔁
## Hello @alicemirigo92  👋 🙂

You're doing a great job on your project. We're proud of you. Good job so far! 

It's OK to have bugs in your code. Please, see this code review as a chance for you to learn practical coding skills and learn new tips that can improve your code quality and help you get a better job. 

There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
  ✅  Correct use of GitHub Flow
  ✅  Descriptive Pull Request
  ✅  No linter errors

## Required Changes ♻️

- [ ] 



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding! 👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/post.rb (hunk 1 above)
Comment (alicemirigo92) [reply app/models/post.rb:8]: @zainsadaqat Hey this is implemented

**Comment by codecaiine** [comment]
Hi @alicemirigo92 I sent you a message on slack can you relpy ?

**Review by codecaiine — APPROVED** [review]
Hi @alicemirigo92 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Good model validations setup ✔️ 

## Optional suggestions
- [ ] [OPTIONAL] I noticed that some of your tests failed since the previous reviewer did not talk about it I would strongly recommend you fix the errors and make all your tests pass in order to make your project works as expected 👍  

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@codecaiine) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**